### PR TITLE
Add Data-to-Pinecone quickstart

### PR DESCRIPTION
**Pull request by chandlerprall**
Here is the first pass!

I've tested each step individually and groups of steps as much as I can between the local and cloud option permutations.

There's some needed edits of the README to make it flow better, but my brain is done thinking about it for today and it's ready to share.

### CODE REVIEW

File: data_to_pinecone_llm/infra/airbyte/main.tf [hunk 1]
```diff
@@ -0,0 +1,98 @@
+// Airbyte Terraform provider documentation: https://registry.terraform.io/providers/airbytehq/airbyte/latest/docs
+
+// Sources
+resource "airbyte_source_bigquery" "bigquery" {
+    configuration = {
+      credentials_json = "...my_credentials_json..."
+      dataset_id       = "...my_dataset_id..."
+      project_id       = "...my_project_id..."
```
Comment (aaronsteers) [inline data_to_pinecone_llm/infra/airbyte/main.tf:8]: These can all be set as terraform variables in `variables.tf`, which will cause them to be prompted at runtime if not provided elsewhere.

Terraform can also add a friendly prompt corresponding to a `description` of the variable, and support secrets by adding a `sensitive = true` declaration inside the variable declaration.

E.g.:

```hcl
variable "db_username" {
  description = "Database administrator username"
  type        = string
  sensitive   = true
}
```

From: https://developer.hashicorp.com/terraform/tutorials/configuration-language/sensitive-variables#refactor-database-credentials

File: data_to_pinecone_llm/README.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+   cd dbt_project
+   ```
+
+2. **Update Connection Details**:
+
+   You'll find a `profiles.yml` file within the directory. This file contains configurations for dbt to connect with your data platform. Update this file with your BigQuery connection details.
+
+3. **Utilize Environment Variables (Optional but Recommended)**:
+
+   To keep your credentials secure, you can leverage environment variables. An example is provided as the `keyfile` config in `profiles.yml`.
```
Comment (flash1293) [inline data_to_pinecone_llm/README.md:154]: Can you extend this a bit? I'm guessing I need to put the bgiquery credentials in a file an set the env variable to the path of the file, but specifics would be nice to follow along mindlessly

File: data_to_pinecone_llm/README.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
+# Data-to-Pinecone Integration
+
+Welcome to the "Data-to-Pinecone Integration" repository! This repo provides a quickstart template for building a full data stack using Airbyte, Notion, Dagster, dbt, BigQuery, and Pinecone. Easily extract data from Notion pages, load it into BigQuery, transform into text blobs using dbt, and publish into Pinecone's vector store so it can be queried or interacted with through an LLM.
```
Comment (flash1293) [inline data_to_pinecone_llm/README.md:3]: A little diagram would help a lot here to explain how all the parts are interacting with each other.

File: data_to_pinecone_llm/infra/airbyte/main.tf [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    }
+    indexing = {
+      index                = "...my_index..."
+      pinecone_environment = "...my_pinecone_environment..."
+      pinecone_key         = "...my_pinecone_key..."
+    }
+    processing = {
+      chunk_overlap = 16
+      chunk_size    = 1024
+      metadata_fields = [
```
Comment (flash1293) [inline data_to_pinecone_llm/infra/airbyte/main.tf:64]: These fields look like they are still following the Slack example

File: data_to_pinecone_llm/infra/airbyte/main.tf [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        ]
+    }
+}
+resource "airbyte_connection" "bigquery_to_pinecone" {
+    name = "BigQuery to Pinecone"
+    source_id = airbyte_source_bigquery.bigquery.source_id
+    destination_id = airbyte_destination_pinecone.pinecone.destination_id
+    configurations = {
+        streams = [
+            { name = "notion" }
```
Comment (flash1293) [inline data_to_pinecone_llm/infra/airbyte/main.tf:95]: This looks like it's syncing the full dataset every time - not a blocker, but could we get an incremental sync in here? Would be a pretty nice showcase

File: data_to_pinecone_llm/README.md (hunk 3 above)
Comment (aaronsteers) [reply data_to_pinecone_llm/README.md:3]: In regards to diagrams, you can copy/paste from here if helpful: https://whimsical.com/from-etl-to-eltp-ai-practitioners-LyCBYNCKbASMXKVFozao4Y@7YNFXnKbYwBVGcyUB3Vg9

There aren't any diagrams that exactly match this flow, but I have several different flows diagramed - you can pick the closest one, duplicate it, and customize for this example.

**Review by aaronsteers — CHANGES_REQUESTED** [review]
This is awesome, thanks, @chandlerprall! 

I left a few comments and I know Joe did as well. 

The highest pri comment from my side is just the handling of variables and secrets in Terraform. Can we move those to terraform variables, with the `sensitive` flag on those that will need to be kept out of git history? (Optionally, we can create a default `.tfvars` file for the user's convenience to fill in all in one place.)

I may have more feedback as I go in today and repro the projects in prep for next week's webinar. But to be clear, I'm not blocked by anything. And as mentioned, I'll create a branch immediately so I can make changes/fixes to feed back to this branch and/or to the main branch.

Great work! 🚀

File: data_to_pinecone_llm/infra/airbyte/main.tf (hunk 1 above)
Comment (chandlerprall) [reply data_to_pinecone_llm/infra/airbyte/main.tf]: Nice, thanks! I've moved most configuration items into variables & made keys/tokens/secrets sensitive

File: data_to_pinecone_llm/README.md [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+
+### Notion
+
+We'll source Notion pages to make the content searchable in Pinecone. Follow the [Notion source docs](https://docs.airbyte.com/integrations/sources/notion) for information on configuring a Notion source.
+
+### BigQuery
+
+BigQuery will store the raw API data from our sources and also the transformed data from dbt. You'll need a BigQuery project and a dataset with a service account that can control the dataset. Airbyte's [BigQuery destination docs](https://docs.airbyte.com/integrations/destinations/bigquery) lists the requirements and links describing how to configure.
+
+### Pinecone index
```
[the remark this answers is not in the document]
Comment (chandlerprall) [reply data_to_pinecone_llm/README.md]: Whoops, done & done

File: data_to_pinecone_llm/infra/airbyte/main.tf (hunk 4 above)
Comment (chandlerprall) [reply data_to_pinecone_llm/infra/airbyte/main.tf]: I was optimistic about getting Slack in there :-D

Removed for now, still need to get the last edited timestamp working on the notion data

**Comment by chandlerprall** [comment]
High-priority items should be addressed; I'll fill in a better description & get one or more diagrams in place as well.

Also high on my list: incremental updates from notion data, add google docs

**Comment by chandlerprall** [comment]
* ~incremental updates~ [`46e7dcf`](https://github.com/airbytehq/quickstarts/pull/83/commits/46e7dcfb3cdec6a1e3e6a45c354b509f3e583e97)

> When was the Security & Privacy page last modified?
>  The Security & Privacy page was last modified on 2023-11-02T23:21:00Z, according to the Notion page (https://www.notion.so/Security-Privacy-5...3).

File: data_to_pinecone_llm/README.md [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+
+   ```bash
+   cd dbt_project
+   ```
+
+2. **Update Connection Details**:
+
+   You'll find a `profiles.yml` file within the directory. This file contains configurations for dbt to connect with your data platform. Update this file with your BigQuery connection details.
+
+3. **Utilize Environment Variables (Optional but Recommended)**:
```
[the remark this answers is not in the document]
Comment (chandlerprall) [reply data_to_pinecone_llm/README.md:168]: I think this part of the setup was refactored and my comment doesn't apply in this regard; will update!

File: data_to_pinecone_llm/README.md (hunk 2 above)
Comment (aaronsteers) [reply data_to_pinecone_llm/README.md:170]: I just merged my changes in from the other PR (#85).

The profiles.yml now references env vars, and the env vars are expectd in `.env`, which can be created by copying from `.env.template`.

**Comment by chandlerprall** [comment]
Pushed what I believe to be the final form of things. Re-tested from start to finish and ensured all of the configurations align.

**Review by aaronsteers — APPROVED** [review]

**Pull request merged**